src/Base.ts

```diff
@@ -0,0 +1,174 @@
+import { computed } from 'mobx';
+import * as _ from 'lodash';
+
+import {
+  $try,
+  $isEvent,
+  hasIntKeys } from './utils';
+import Actions from './shared/Actions';
+import Events from './shared/Events';
+import Helpers from './shared/Helpers';
+import Initializer from './shared/Initializer';
+import Utils from './shared/Utils';
+
+export default abstract class Base implements Actions, Events, Helpers, Initializer, Utils {
+  abstract invalidate(message?: any, async?: boolean):any;
```

Comment:
If we don't put these in as abstract, nothing compiles. It's kind of messy but does work.